File: 구름 알고리즘 미션/기초 시뮬레이션/나무꾼 구름이/나무꾼 구름이.py
N, M, xs = map(int, input().split(" "))
H = list(map(int, input().split(" ")))
Q = int(input())
D = list(map(str, input().split(" ")))

result = 0
pos = xs - 1
for idx in range(Q):
  if H[pos] >= M:
    result += H[pos]
    H[pos] = 0
  
  H = [x+1 for x in H]

  if D[idx] == 'L':
    pos = (pos-1+N) % N
  elif D[idx] == 'R':
    pos = (pos+1) % N
print(result)

File: 구름 알고리즘 미션/기초 시뮬레이션/나무꾼 구름이/나무꾼 구름이.js
// Run by Node.js
const readline = require('readline');

(async () => {
	let rl = readline.createInterface({ input: process.stdin });
	let input = [];
	
	for await (const line of rl) {
		input.push(line);
		if(input.length === 4) rl.close();
	}
	
	const [N, M, Xs] = input[0].split(" ").map(Number);
	let H = input[1].split(" ").map(Number);
	const Q = +input[2];
	const D = input[3].split(" ");
	
	let result = 0;
	let pos = Xs - 1;
	
	for(let i = 0 ; i < Q ; i++){
		if(H[pos] + i >= M){
			result += H[pos] + i;
			H[pos] -= H[pos] + i;
		}
		
		if(D[i] === 'L') pos = (pos-1 + N) % N;
		else if(D[i] === 'R') pos = (pos+1) % N;
	}
	
	console.log(result);
	
	process.exit();
})();
